packages/legend-query-builder/src/components/__tests__/QueryBuilderFilterPanel.test.tsx

```diff
@@ -2699,199 +2694,575 @@ test(
+      renderResult.getByText('Type').parentElement,
+    );
+    selectFromCustomSelectorInput(typeContainer, 'Number');
+    constantValueInput = getConstantValueInput(renderResult);
+    fireEvent.change(constantValueInput, { target: { value: '5' } });
+    fireEvent.click(renderResult.getByRole('button', { name: 'Apply' }));
+
+    // Verify 1 validation error
     expect(getByText(filterPanel, '1 issue')).not.toBeNull();
+    expect(
+      getByTitle(
+        filterPanel,
+        'Filter value for Legal Name is missing or invalid',
+        { exact: false },
+      ),
+    ).not.toBeNull();
     expect(
       renderResult.getByRole('button', { name: 'Run Query' }),
     ).toHaveProperty('disabled', true);
 
-    // Select value
-    const filterValueButton = getByTitle(
-      filterPanel,
-      'Click to edit and pick from more date options',
+    // Create parameter of type string
+    const parametersPanel = renderResult.getByTestId(
+      QUERY_BUILDER_TEST_ID.QUERY_BUILDER_PARAMETERS,
     );
-    fireEvent.click(filterValueButton);
-    fireEvent.click(renderResult.getByText(CUSTOM_DATE_PICKER_OPTION.TODAY));
-    fireEvent.keyDown(
-      renderResult.getByText(CUSTOM_DATE_PICKER_OPTION.ABSOLUTE_DATE),
-      {
-        key: 'Escape',
-        code: 'Escape',
-      },
+    fireEvent.click(getByTitle(parametersPanel, 'Add Parameter'));
+    const parameterNameInput = getParameterNameInput(renderResult);
+    fireEvent.change(parameterNameInput, { target: { value: 'p_var_1' } });
+    fireEvent.click(renderResult.getByRole('button', { name: 'Create' }));
+
+    // Drag and drop parameter to filter panel value
+    const parameterDragSource = await findByText(parametersPanel, 'p_var_1');
+    await dragAndDrop(
+      parameterDragSource,
+      filterPanel,
+      filterPanel,
+      'Change Filter Value',
     );
 
-    // Verify no validation issues
-    expect(getByText(filterPanel, '"Today"')).not.toBeNull();
+    // Verify no validation error
     expect(queryByText(filterPanel, '1 issue')).toBeNull();
     expect(
       renderResult.getByRole('button', { name: 'Run Query' }),
     ).toHaveProperty('disabled', false);
+
+    // Change parameter type
+    fireEvent.click(getByText(parametersPanel, 'p_var_1'));
+    typeContainer = guaranteeNonNullable(
+      renderResult.getByText('Type').parentElement,
+    );
+    selectFromCustomSelectorInput(typeContainer, 'Number');
+    fireEvent.click(renderResult.getByRole('button', { name: 'Update' }));
+
+    // Verify 1 validation error
+    expect(getByText(filterPanel, '1 issue')).not.toBeNull();
+    expect(
+      getByTitle(
+        filterPanel,
+        'Filter value for Legal Name is missing or invalid',
+        { exact: false },
+      ),
+    ).not.toBeNull();
+    expect(
+      renderResult.getByRole('button', { name: 'Run Query' }),
+    ).toHaveProperty('disabled', true);
   },
 );
 
+type FilterPanelDefaultValueTestCase = {
```

Comment:
Combined all the null default value tests into one `test.each` generic test, similar to what I did for the `QueryBuilderPostFilterPanel` null value tests.